packages/@contentlayer/cli/src/commands/_BaseCommand.ts

```diff
@@ -21,26 +20,22 @@ export abstract class BaseCommand extends Command {
     description: 'More verbose logging and error stack traces',
   })
 
-  async execute() {
-    try {
-      if (this.clearCache) {
-        await fs.rm(core.ArtifactsDir.getDirPath({ cwd: process.cwd() }), { recursive: true })
-        console.log('Cache cleared successfully')
-      }
-
-      await pipe(
-        this.executeSafe,
-        T.provideSomeLayer(JaegerNodeTracing('contentlayer-cli')),
-        T.tapCause((cause) => (this.verbose ? T.die(pretty(cause)) : T.unit)),
-        T.runPromise,
-      )
-    } catch (e: any) {
-      if (e._tag !== 'HandledFetchDataError') {
-        console.error(errorToString(e))
-      }
-      process.exit(1)
-    }
-  }
-
   abstract executeSafe: T.Effect<OT.HasTracer & HasClock, unknown, void>
+
+  execute = () =>
+    pipe(
+      pipe(clearCacheIfNeeded(this.clearCache), T.zipRight(this.executeSafe)),
+      core.runMain({
+        tracingServiceName: 'contentlayer-cli',
+        verbose: this.verbose || process.env.CL_DEBUG !== undefined,
+      }),
+    )
 }
+
+const clearCacheIfNeeded = (shouldClearCache: boolean) =>
+  T.gen(function* ($) {
+    if (shouldClearCache) {
+      yield* $(T.promise(() => fs.rm(core.ArtifactsDir.getDirPath({ cwd: process.cwd() }), { recursive: true })))
```

Comment:
nit: you can use the new `rm` module introduced in https://github.com/contentlayerdev/contentlayer/pull/19